tests/ui/invalid_pymethods.stderr

```diff
@@ -88,55 +88,89 @@ error: `signature` not allowed with `classattr`
 105 |     #[pyo3(signature = ())]
     |            ^^^^^^^^^
 
-error: cannot combine these method types
+error: `#[classattr]` may not be combined with `#[staticmethod]`
    --> tests/ui/invalid_pymethods.rs:112:7
     |
 112 |     #[staticmethod]
     |       ^^^^^^^^^^^^
 
+error: `#[new]` does not take any arguments
+       = help: did you mean `#[new] #[pyo3(signature = ())]`?
+   --> tests/ui/invalid_pymethods.rs:118:7
+    |
+118 |     #[new(signature = ())]
+    |       ^^^
+
+error: `#[new]` does not take any arguments
```

Comment:
We should add some additional clarification, something like:
```suggestion
error: `#[new]` does not take any arguments
note = this was previously accepted and ignored
```

Reply:
Good idea, I like that, will add.